source/common/http/http1/codec_impl.cc

```diff
@@ -370,19 +395,25 @@ ConnectionImpl::ConnectionImpl(Network::Connection& connection, Stats::Scope& st
 void ConnectionImpl::completeLastHeader() {
   ENVOY_CONN_LOG(trace, "completed header: key={} value={}", connection_,
                  current_header_field_.getStringView(), current_header_value_.getStringView());
+
   if (!current_header_field_.empty()) {
     toLowerTable().toLowerCase(current_header_field_.buffer(), current_header_field_.size());
     current_header_map_->addViaMove(std::move(current_header_field_),
                                     std::move(current_header_value_));
   }
+
   // Check if the number of headers exceeds the limit.
   if (current_header_map_->size() > max_headers_count_) {
     error_code_ = Http::Code::RequestHeaderFieldsTooLarge;
     sendProtocolError();
     throw CodecProtocolException("headers size exceeds limit");
   }
 
-  header_parsing_state_ = HeaderParsingState::Field;
+  if (processing_trailers_) {
```

Comment:
Is it possible to just reuse `header_parsing_state_` and avoid separate booleans for the trailer states? It seems like we can only be doing headers or trailers and it's clear when we are resetting?

It seems like you might still need `processing_trailers_` to differentiate between dispatching headers and trailers but in general any work you can do to limit the state variables would be good.